app/controllers/api/internal/v1/payment_controller.rb

```diff
@@ -3,65 +3,31 @@
 class Api::Internal::V1::PaymentController < Api::Internal::V1::ApiController
   # We are using our own authentication method with vault
   protect_from_forgery except: [:init_stripe]
+
   def init_stripe
     attendee_id = params.require(:attendee_id)
-    registration_service_user = params.require(:current_user)
-    iso_amount = params.require(:amount)
-
-    holder = AttendeePaymentRequest.create(attendee_id: attendee_id)
-    competition_id, user_id = holder.competition_and_user_id
-
-    competition = holder.competition
-    user = holder.user
-    payee = User.find(registration_service_user)
-    render json: { error: "Registration not found" }, status: :not_found unless competition.present? && user.present? && payee.present?
-    account_id = competition.payment_account_for(:stripe).account_id
+    competition_id, registering_user_id = attendee_id.split("-")
 
-    registration_metadata = {
-      competition: competition.name,
-      registration_url: edit_registration_path(competition_id, user_id),
-    }
+    paying_user_id = params.require(:current_user)
+    paying_user = User.find(paying_user_id)
 
-    currency_iso = params.require(:currency_code)
-    stripe_amount = StripeRecord.amount_to_stripe(iso_amount, currency_iso)
-
-    payment_intent_args = {
-      amount: stripe_amount,
-      currency: currency_iso,
-      receipt_email: user.email,
-      description: "Registration payment for #{competition.name}",
-      metadata: registration_metadata,
-    }
+    competition = Competition.find(competition_id)
+    render json: { error: "Competition not found" }, status: :not_found unless competition_id.present?
```

Comment:
Should this not go above Line 14? Given that we're trying to use competition_id in line 14

Reply:
Ah that makes!